hnn_core/optimization.py

```diff
@@ -60,52 +69,47 @@ def _split_by_evinput(params, sigma_range_multiplier, timing_range_multiplier,
     """
 
     evinput_params = {}
-    for evinput_t in params['t_*']:
-        id_str = evinput_t.lstrip('t_')
-        timing_mean = float(params[evinput_t])
-
-        if f'sigma_{evinput_t}' in params:
-            timing_sigma = float(params['sigma_' + evinput_t])
-        else:
-            timing_sigma = 3.0
-            print("Couldn't fing timing_sigma. Using default %f" %
-                  timing_sigma)
+    for drive_idx, drive_name in enumerate(drive_names):
+        timing_mean = drive_dynamics[drive_idx]['mu']
+        timing_sigma = drive_dynamics[drive_idx]['sigma']
 
         if timing_sigma == 0.0:
             # sigma of 0 will not produce a CDF
             timing_sigma = 0.01
 
-        evinput_params[id_str] = {'mean': timing_mean, 'sigma': timing_sigma,
-                                  'ranges': {}}
+        evinput_params[drive_name] = {'mean': timing_mean,
+                                      'sigma': timing_sigma,
+                                      'ranges': {}}
 
-        evinput_params[id_str]['ranges']['sigma_' + evinput_t] = \
+        evinput_params[drive_name]['ranges'][f'{drive_name}_sigma'] = \
             _get_range(timing_sigma, sigma_range_multiplier)
 
         # calculate range for time
         timing_bound = timing_sigma * timing_range_multiplier
         range_min = max(0, timing_mean - timing_bound)
-        range_max = min(float(params['tstop']), timing_mean + timing_bound)
+        range_max = min(tstop, timing_mean + timing_bound)
 
-        evinput_params[id_str]['start'] = range_min
-        evinput_params[id_str]['end'] = range_max
-        evinput_params[id_str]['ranges'][evinput_t] =  \
+        evinput_params[drive_name]['start'] = range_min
+        evinput_params[drive_name]['end'] = range_max
+        evinput_params[drive_name]['ranges'][f'{drive_name}_mu'] = \
             {'initial': timing_mean, 'minval': range_min, 'maxval': range_max}
 
         # calculate ranges for syn. weights
-        for label in params[f'gbar_{id_str}*']:
-            value = params[label]
-            ranges = _get_range(value, synweight_range_multiplier)
-            if value == 0.0:
-                ranges['minval'] = value
+        for syn_weight_key in drive_syn_weights[drive_idx]:
+            new_key = f'{drive_name}_gbar_{syn_weight_key}'
+            weight = drive_syn_weights[drive_idx][syn_weight_key]
+            ranges = _get_range(weight, synweight_range_multiplier)
+            if weight == 0.0:
```

Comment:
What happens when `weight != 0.0`?

Reply:
This is something Blake originally built in as a general upper bound for synaptic weight values. It's somewhat arbitrary, but I've very rarely seen or tuned weight values above 1.0 as this is orders of magnitude larger than most of the other weights in previous param files.

Reply:
Sounds good! Truthfully there probably isn't a practical upper bound when we consider that the effect of increased conductance diminishes (especially once a single input is enough to evoke an action potential)

Reply:
We should def revisit this. Maybe we can establish some sort of convention that we can use for both optimization and SBI that 1) constrains our parameter space with reasonable boundary conditions and 2) isn't too arbitrary.